styx-common/src/main/java/com/spotify/styx/util/WorkflowValidator.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * -\-\-
+ * Spotify Styx Scheduler Service
+ * --
+ * Copyright (C) 2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.styx.util;
+
+import static java.lang.String.format;
+
+import com.spotify.styx.model.Workflow;
+import com.spotify.styx.model.WorkflowConfiguration;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class WorkflowValidator {
+
+  static final int MAX_ID_LENGTH = 256;
+  static final int MAX_DOCKER_ARGS_TOTAL = 1024;
+  static final int MAX_RESOURCES = 5;
+  static final int MAX_RESOURCE_LENGTH = 256;
+  static final int MAX_COMMIT_SHA_LENGTH = 256;
+  static final int MAX_SECRET_NAME_LENGTH = 253;
+  static final int MAX_SECRET_MOUNT_PATH_LENGTH = 4096;
```

Comment:
is this number backed up by some max length of dir path?

Reply:
https://github.com/torvalds/linux/blob/master/include/uapi/linux/limits.h#L13 =)